src/Core/ServiceCloner/ServiceClonerService.php

```diff
@@ -0,0 +1,165 @@
+
+    /**
+     * @var ServiceClonerStateService
+     */
+    private $serviceClonerStateService;
+
+    public function __construct(
+        ConfigurationServiceInterface $dockerConfiguration,
+        LoggerInterface $logger,
+        Filesystem $filesystem,
+        FilesystemServiceInterface $zfs,
+        ContainerCreationServiceInterface $dockerService,
+        ServiceClonerLifeCycleHookServiceInterface $serviceClonerLifeCycleHookService,
+        ServiceClonerNamingServiceInterface $serviceClonerNamingService,
+        ServiceClonerStateService $serviceClonerStateService,
+        ContainerStateServiceInterface $dockerStateService,
+        SluggerInterface $slugger
+    ) {
+        $this->dockerConfiguration = $dockerConfiguration;
+        $this->configuration = $this->dockerConfiguration->getConfiguration();
+        $this->logger = $logger;
+        $this->filesystem = $filesystem;
+        $this->zfsService = $zfs;
+        $this->slugger = $slugger;
+        $this->dockerService = $dockerService;
+        $this->dockerStateService = $dockerStateService;
+        $this->serviceClonerLifeCycleHookService = $serviceClonerLifeCycleHookService;
+        $this->serviceClonerNamingService = $serviceClonerNamingService;
+        $this->serviceClonerStateService = $serviceClonerStateService;
+    }
+
+    public function start(string $masterName, string $instanceName, int $index): void
+    {
+        dump(sprintf('-------------%s----------', $this->serviceClonerNamingService->getFullName($masterName, $instanceName, '@')));
+        $this->startZfs($masterName, $instanceName);
+        $this->startDocker($masterName, $instanceName, $index);
+        $this->serviceClonerStateService->saveState($masterName, $instanceName, $index);
+    }
+
+    private function startDocker(string $masterName, string $instanceName, int $index): void
+    {
+        $containerName = $this->serviceClonerNamingService->getFullName($masterName, $instanceName, '_');
+
+        $containerParameter = new ContainerParameterDTO(
+            $containerName,
+            $index,
+            $this->serviceClonerNamingService->getZfsFilesystemPath($masterName, $instanceName)
+        );
+
+        if ($this->dockerStateService->dockerState($containerName) === 'running') {
+            return;
+        }
+
+        $service = $this->configuration->getServiceByName($masterName);
+        if ($service === null) {
+            throw new NonExistingServiceException($masterName);
+        }
+
+        $this->serviceClonerLifeCycleHookService->preStart($service, $containerParameter);
+        $this->dockerService->createDocker(
+            $containerParameter,
+            $service
+        );
+        $this->serviceClonerLifeCycleHookService->postStartWaiter($service, $containerParameter);
+        $this->serviceClonerLifeCycleHookService->postStart($service, $containerParameter);
+    }
+
+    private function startZfs(string $masterName, string $instanceName): void
```

Comment:
```suggestion
    private function startFilesystem(string $masterName, string $instanceName): void
```
depending on the answers to the previous questions